codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/rustlang/RustTypes.kt

```diff
@@ -77,16 +78,17 @@ sealed class RustType {
     data class HashSet(override val member: RustType) : RustType(), Container {
         override val name = Type
         override val namespace = Namespace
-
         companion object {
-            // This is Vec intentionally. Note the following passage from the Smithy spec:
+            // Note the following passage from the Smithy spec:
             //    Sets MUST be insertion ordered. Not all programming languages that support sets
             //    support ordered sets, requiring them may be overly burdensome for users, or conflict with language
             //    idioms. Such languages SHOULD store the values of sets in a list and rely on validation to ensure uniqueness.
-            // It's possible that we could provide our own wrapper type in the future.
-            const val Type = "Vec"
-            const val Namespace = "std::vec"
-            val RuntimeType = RuntimeType(name = Type, namespace = Namespace, dependency = null)
+            const val Type = "Set"
+            const val Namespace = "aws_smithy_types"
+
+            fun RuntimeType(runtimeConfig: RuntimeConfig): RuntimeType {
```

Comment:
We've promoted this from a `val` to a function in order to accommodate the need for a `dependency`.